aiozk/test/test_watchers.py

```diff
@@ -261,3 +262,51 @@ def cb(resp):
 
     assert len(out) == num * 2
     await zk.deleteall(path)
+
+
+@pytest.mark.asyncio
+async def test_data_watcher_no_race_condition(zk, path):
+    """
+    Ensure that there is no race condition between a task setting watch-events
+    and a task waiting watch-events.
+
+    A watch-event can be triggered right after it is set on a znode while there
+    are no callbacks for the event so that the event is ignored quietly. And
+    then a successive call to wait for watch-event will never be finished.
+    """
+
+    class DelayedDataWatcher(DataWatcher):
+        def __init__(self, *args, **kwargs):
+            super().__init__(*args, **kwargs)
+
+        async def fetch(self, path):
+            await super().fetch(path)
+            await asyncio.sleep(0.2)
+
+    await zk.create(path)
+    data_watcher = DelayedDataWatcher()
+    data_watcher.set_client(zk)
+
+    async def callback(d):
+        await asyncio.sleep(0)
+
+    data_watcher.add_callback(path, callback)
+    await asyncio.sleep(0.1)
+
+    await zk.set_data(path, uuid.uuid4().hex)
+    await asyncio.sleep(1)
+
+    ready = asyncio.Event()
+
+    def set_ready(_):
+        ready.set()
+
+    data_watcher.add_callback(path, set_ready)
+    await zk.set_data(path, uuid.uuid4().hex)
+
+    await asyncio.wait_for(ready.wait(), timeout=1)
+    assert ready.is_set()
+
+    await asyncio.sleep(0.2)
+
+    await zk.delete(path)
```

Comment:
```    
    data_watcher.remove_callback(path, callback)
    data_watcher.remove_callback(path, set_ready)
```
Should be here or tests will fail with error about non-finished (`Task was destroyed but it is pending!`)
I'm testing with 
```./venv/bin/pytest aiozk/test/test_watchers.py::test_watcher_fires_after_nonode aiozk/test/test_watchers.py::test_data_watcher_no_race_condition -s --log-cli-level DEBUG```

Reply:
@cybergrind I've just reflected your advice and modified to call remove_callback() methods. Thanks.

Reply:
But without calling remove_callback() method, test did not fail even if I ran pytest several times.

Reply:
That's why I missed to call remove_callback().

Reply:
@rhdxmr on my machine it was failing 100% of time with and without a fix.